File: model-serving/src/model_serving/logger.py
"""Structured logging for the model-serving deployable.

Deliberately generic and content-free at M0: this is transport/format
configuration only, mirroring the orchestrator's logger.ts so both
deployables produce comparable structured logs. Any future
embedding/classifier/ASR logging must go through this shared logger rather
than bare print statements.
"""

from __future__ import annotations

import logging
import sys

from model_serving.config import ModelServingConfig


class _JsonLikeFormatter(logging.Formatter):
    def format(self, record: logging.LogRecord) -> str:
        base = (
            f'{{"level":"{record.levelname.lower()}",'
            f'"time":"{self.formatTime(record, "%Y-%m-%dT%H:%M:%S%z")}",'
            f'"name":"{record.name}",'
            f'"msg":"{record.getMessage()}"}}'
        )
        return base


def create_logger(config: ModelServingConfig) -> logging.Logger:
    logger = logging.getLogger(config.service_name)
    logger.setLevel(config.log_level.upper())
    logger.propagate = False

    if not logger.handlers:
        handler = logging.StreamHandler(stream=sys.stdout)
        handler.setFormatter(_JsonLikeFormatter())
        logger.addHandler(handler)

    return logger

File: orchestrator/src/logger.ts
import pino from 'pino';

import type { OrchestratorConfig } from './config.js';

/**
 * Structured logging for the orchestrator deployable.
 *
 * Deliberately generic and content-free at M0: this is transport/format
 * configuration only. Any future signal-health, evidence, or lifecycle
 * logging must go through this shared logger rather than console.* so that
 * log level, format, and (later) redaction policy stay centrally governed.
 */
export function createLogger(config: OrchestratorConfig) {
  return pino({
    name: config.serviceName,
    level: config.logLevel,
    formatters: {
      level(label) {
        return { level: label };
      },
    },
    timestamp: pino.stdTimeFunctions.isoTime,
  });
}

export type Logger = ReturnType<typeof createLogger>;
